### ISSUE TITLE
rule for detecting format error in gettext translations

### ISSUE THREAD

**Issue by spaceone** [charliermarsh/ruff#3294]
```
import gettext
_ = gettext.NullTranslations().gettext
var = 'baz'
_('foo %s bar' % var)
_('foo {} bar'.format(var))
_(f'foo {var} bar')
```

those 3 cases should be detected in a new rule.

**Comment by charliermarsh** [comment]
Is the error that you're _supposed_ to leave the parameters unfilled?

**Comment by charliermarsh** [comment]
(Seems like a reasonable rule, I'm just not familiar with the API.)

**Comment by Jackenmen** [comment]
The error is that you're supposed to use constant strings, not dynamic ones as that passed string is used as a key during the lookup in message catalog.

I think that ideally this should support not just the most basic gettext call accepting a single argument but also calls to gettext, ngettext, and so on:
https://docs.python.org/3/library/gettext.html

**Comment by spaceone** [comment]
thanks for the feedback, I clarified the initial description.

**Comment by charliermarsh** [comment]
Thank you!
